pallets/pallet-bonded-coins/src/tests/transactions/create_pool.rs

```diff
@@ -1 +1,251 @@
+
+			let bonded_token = TokenMetaOf::<Test> {
+				name: BoundedVec::truncate_from(b"Bitcoin".to_vec()),
+				symbol: BoundedVec::truncate_from(b"btc".to_vec()),
+				min_balance: 1,
+			};
+
+			let bonded_tokens = vec![bonded_token; 3];
+
+			assert_eq!(bonded_tokens.len(), 3);
+
+			assert_ok!(BondingPallet::create_pool(
+				origin,
+				curve,
+				DEFAULT_COLLATERAL_CURRENCY_ID,
+				BoundedVec::truncate_from(bonded_tokens),
+				10,
+				true
+			));
+
+			assert_eq!(NextAssetId::<Test>::get(), 3);
+
+			let pool_id = calculate_pool_id(&[0, 1, 2]);
+
+			let details = Pools::<Test>::get(pool_id).unwrap();
+
+			assert_eq!(BondingPallet::get_currencies_number(&details), 3);
+			assert_eq!(details.bonded_currencies, vec![0, 1, 2]);
+
+			assert_eq!(
+				Balances::free_balance(ACCOUNT_00),
+				initial_balance.sub(BondingPallet::calculate_pool_deposit(3))
+			);
+		});
+}
+
+#[test]
+fn can_create_identical_pools() {
+	let initial_balance = 100_000_000_000_000_000u128;
+	ExtBuilder::default()
+		.with_native_balances(vec![(ACCOUNT_00, initial_balance)])
+		.with_collaterals(vec![DEFAULT_COLLATERAL_CURRENCY_ID])
+		.build()
+		.execute_with(|| {
+			assert_eq!(NextAssetId::<Test>::get(), 0);
+
+			let origin: OriginFor<Test> = RawOrigin::Signed(ACCOUNT_00).into();
+			let curve = get_linear_bonding_curve_input();
+
+			let bonded_token = TokenMetaOf::<Test> {
+				name: BoundedVec::truncate_from(b"Bitcoin".to_vec()),
+				symbol: BoundedVec::truncate_from(b"btc".to_vec()),
+				min_balance: 1,
+			};
+
+			assert_ok!(BondingPallet::create_pool(
+				origin.clone(),
+				curve.clone(),
+				DEFAULT_COLLATERAL_CURRENCY_ID,
+				BoundedVec::truncate_from(vec![bonded_token.clone()]),
+				10,
+				true
+			));
+
+			assert_ok!(BondingPallet::create_pool(
+				origin,
+				curve,
+				DEFAULT_COLLATERAL_CURRENCY_ID,
+				BoundedVec::truncate_from(vec![bonded_token]),
+				10,
+				true
+			));
+
+			assert_eq!(NextAssetId::<Test>::get(), 2);
+
+			let details1 = Pools::<Test>::get(calculate_pool_id(&[0])).unwrap();
+			let details2 = Pools::<Test>::get(calculate_pool_id(&[1])).unwrap();
+
+			assert_eq!(details1.bonded_currencies, vec![0]);
+			assert_eq!(details2.bonded_currencies, vec![1]);
+		});
+}
+
+#[test]
+fn fails_if_collateral_not_exists() {
+	ExtBuilder::default()
+		.with_native_balances(vec![(ACCOUNT_00, 100_000_000_000_000_000u128)])
+		.build()
+		.execute_with(|| {
+			let origin = RawOrigin::Signed(ACCOUNT_00).into();
+			let curve = get_linear_bonding_curve_input();
+
+			let bonded_token = TokenMetaOf::<Test> {
+				name: BoundedVec::truncate_from(b"Bitcoin".to_vec()),
+				symbol: BoundedVec::truncate_from(b"btc".to_vec()),
+				min_balance: 1,
+			};
+
+			assert_err!(
+				BondingPallet::create_pool(
+					origin,
+					curve,
+					DEFAULT_COLLATERAL_CURRENCY_ID,
+					BoundedVec::truncate_from(vec![bonded_token]),
+					10,
+					true
+				),
+				AssetsPalletErrors::<Test>::Unknown
```

Comment:
Sorry I have missed follow-up on previous conversations. While the `Unknown` error here if a collateral does not exist? Shouldn't the error be more specific?

Reply:
I also tripped here for a minute, but it's the assets pallet's `Unknown` error, which is meant to express that an asset is unknown:
```rust
		/// The given asset ID is unknown.
		Unknown,
```
Weird choice indeed because it looks like an 'unknown error', but when it shows up in our runtime as `Fungibles:Unknown` in combination with the docstring associated with it (which is carried over to the metadata afaik) it should become clearer.